exercise/api/full_serializers.py

```diff
@@ -60,18 +65,26 @@ def get_post_url(self, obj):
             return request.build_absolute_uri(url)
         return None
 
+    def get_embedded_in_name(self, obj: BaseExercise) -> str:
+        parent = obj.parent
+        if isinstance(parent, LearningObject):
+            return str(parent)
+        return ''
+
```

Comment:
`obj.parent` is always either None or a LearningObject, so it is enough to check whether the parent is None. `embedded_in_name` should also be renamed to `parent_name` to match the naming convention everywhere else. I believe you can also return `None` if the parent is None which would be preferable. Is there a reason why this returns `str(parent)` instead of `parent.id` (and name it `parent_id` instead of `parent_name`)? Note that `str(parent)` isn't necessarily unique to the parent learning object.

Reply:
The name of the parent is nearly impossible to get from the API (as the API doesn't have pages for the chapters, only exercises), and at least in Jutut. It doesn't need to be unique, one just needs the title of the chapter. But good points about changing the logic and the name of the field, I'll change those.